apps/addons/management/commands/approve_addons.py

```diff
@@ -0,0 +1,93 @@
+from django.core.management.base import BaseCommand, CommandError
+
+import amo
+from addons.models import Addon
+from editors.helpers import ReviewHelper
+
+
+class Command(BaseCommand):
+    args = '<addon_guid addon_guid ...>'
+    help = 'Approve a list of add-ons, given their GUIDs.'
+
+    def handle(self, *args, **options):
+        if len(args) == 0:  # No GUID provided?
+            raise CommandError(
+                'Please provide at least one add-on guid to approve.')
+
+        files = get_files(args)
+
+        if not files:
+            raise CommandError('No files found that need approval')
+
+        files_with_review_type = [
+            (file_, get_review_type(file_)) for file_ in files]
+
+        for file_, review_type in files_with_review_type:
+            print u'Addon {0}, version {1}, file {2}, review type {3}'.format(
+                file_.version.addon, file_.version, file_, review_type)
+
+        confirm = raw_input(
+            'Are you sure you want to bulk approve and sign all those {0} '
+            'files? (yes/no)'.format(len(files)))
+        if confirm != 'yes':
+            raise CommandError('Aborted.')
+
+        approve_files(files_with_review_type)
+
+
+def get_files(addon_guids):
+    """Return the list of files that need approval, given a list of GUIDs.
+
+    A file needs approval if:
+    - it's unreviewed
+    - it's preliminary reviewed, but its addon is requesting a full review
+    """
+    # Get all the add-ons that have a GUID from the list, and which are either
+    # reviewed or awaiting a review.
+    addons = Addon.with_unlisted.filter(
+        guid__in=addon_guids,
+        status__in=amo.UNDER_REVIEW_STATUSES + amo.REVIEWED_STATUSES)
+
+    # Of all those add-ons, we return the list of latest_version files that are
+    # under review, or add-ons that are under review.
+    return [file_
+            for addon in addons
+            for file_ in addon.latest_version.files.all()
+            if (file_.status in amo.UNDER_REVIEW_STATUSES or
+                addon.status in amo.UNDER_REVIEW_STATUSES)]
+
+
+def approve_files(files_with_review_type):
+    """Approve the files (and sign them).
+
+    A file will be fully approved if:
+    - it's waiting for a full review
+    - it's preliminary reviewed, and its addon is waiting for a full review
+    A file will be prelim approved if:
+    - it's waiting for a prelim review
+    """
+    for file_, review_type in files_with_review_type:
+        version = file_.version
+        addon = version.addon
+        helper = ReviewHelper(request=None, addon=addon,
+                              version=file_.version)
+        # Provide the file to review/sign to the helper.
+        helper.set_data({'addon_files': [file_],
+                         'comments': 'bulk approval'})
+        if review_type == 'full':
+            # Already fully reviewed, or waiting for a full review.
+            helper.handler.process_public()
```

Comment:
Yep, it's all taken care of already, it adds a log stating that it's an approval done by the "mozilla" user, with the comment `bulk approval` from three lines above